test/CommandLineUtils.Tests/CommandLineApplicationTests.cs

```diff
@@ -643,6 +643,40 @@ public void HelpOptionIsSet()
             Assert.True(helpOption.HasValue());
         }
 
+        [Theory]
+        [InlineData("-h")]
+        [InlineData("--help")]
+        public void HelpOptionIsInherited(string helpOptionString)
+        {
+            var app = new CommandLineApplication
+            {
+                Out = new XunitTextWriter(_output)
+            };
+            var helpOption = app.HelpOption(true);
+            var subCommand = app.Command("lvl2", subCmd => { });
+
+            var commandOptions = new[] { "lvl2", helpOptionString };
+            app.Execute(commandOptions);
+            Assert.True(helpOption.HasValue());
```

Comment:
Can you also assert the help output is (1) shown and (2) is the help output for lvl2, not the root command?

Reply:
no problem